server/athenian/api/internal/features/entries.py

```diff
@@ -1516,6 +1511,211 @@ def check_runs_grouper(df: pd.DataFrame) -> list[np.ndarray]:
+class MetricsLineRequest:
+    """Common base for multiple metrics request classes."""
+
+    metrics: Sequence[str]
+    time_intervals: Sequence[Sequence[datetime]]
+    teams: Sequence[TeamSpecificFilters]
+    metric_params: dict[str, Any] = dataclasses.field(default_factory=dict)
+
+    def __str__(self) -> str:
+        """Summarize the request."""
+        return (
+            f"<[{', '.join(self.metrics)}],"
+            f" [{', '.join('[%s]' % ', '.join(str(d) for d in s) for s in self.time_intervals)}],"
+            f" [{', '.join(str(f) for f in self.teams)}]>"
+        )
+
+    def __sentry_repr__(self) -> str:
+        """Format for Sentry reports."""
+        return str(self)
+
+    def all_jira_filters(self) -> Iterator[JIRAFilter]:
+        """Collect all JIRAFilter from the included TeamSpecificFilters."""
+        return (td.jira_filter for td in self.teams)
+
+
+@dataclasses.dataclass(frozen=True, slots=True)
+class TeamSpecificFilters:
+    """Filters that are different for each team."""
+
+    team_id: int
+    # filters start here
+    repositories: tuple[str]
+    participants: PRParticipants | ReleaseParticipants | JIRAParticipants
+    jira_filter: JIRAFilter
+    # add more filters here
+
+    def __str__(self) -> str:
+        """Format the filters as a stable string."""
+        # participants are fully defined by the team ID
+        return f"{self.team_id}|{','.join(self.repositories)}|{self.jira_filter}"
+
+    def __sentry_repr__(self):
+        """Format for Sentry."""
+        dikt = dataclasses.asdict(self)
+        del dikt["participants"]  # fully defined by the team ID
+        return str(dikt)
+
+
+class PRFactsCalculator:
+    """Calculator for Pull Requests facts."""
+
+    _log = logging.getLogger(f"{metadata.__package__}.PRFactsCalculator")
+
+    def __init__(
+        self,
+        account: int,
+        meta_ids: tuple[int, ...],
+        mdb: Database,
+        pdb: Database,
+        rdb: Database,
+        pr_miner: Type[PullRequestMiner] = PullRequestMiner,
+        branch_miner: Type[BranchMiner] = BranchMiner,
+        done_prs_facts_loader: Type[DonePRFactsLoader] = DonePRFactsLoader,
+        unfresh_pr_facts_fetcher: Type[
+            UnfreshPullRequestFactsFetcher
+        ] = UnfreshPullRequestFactsFetcher,
+        pr_jira_mapper: Type[PullRequestJiraMapper] = PullRequestJiraMapper,
+        cache: Optional[aiomcache.Client] = None,
+    ):
+        """Init the `PRFactsCalculator`."""
+        self._account = account
+        self._meta_ids = meta_ids
+        self._mdb = mdb
+        self._pdb = pdb
+        self._rdb = rdb
+        self._pr_miner = pr_miner
+        self._branch_miner = branch_miner
+        self._done_prs_facts_loader = done_prs_facts_loader
+        self._unfresh_pr_facts_fetcher = unfresh_pr_facts_fetcher
+        self._pr_jira_mapper = pr_jira_mapper
+        self._cache = cache
+
     async def calc_pull_request_facts_github(
```

Comment:
```suggestion
    async def __call__(
```
This is logical... This is the only thing this class is supposed to do.